lighthouse-core/gather/drivers/driver.js

```diff
@@ -219,6 +210,28 @@ class Driver {
   }
 
   /**
+   * If our main document URL redirects, we will update options.url accordingly
+   * As such, options.url will always represent the post-redirected URL.
+   * options.initialUrl is the pre-redirect URL that things started with
+   *
+   * Caveat: only works when network recording enabled for a pass
+   */
+  updateUrlIfRedirected(opts) {
+    const networkManager = this._networkRecorder.networkManager;
+    networkManager.addEventListener(this._networkRecorder.EventTypes.RequestFinished, request => {
```

Comment:
should add an event listener interface on `NetworkRecorder` so we don't have to reach into it here?